velox/exec/tests/ExpandTest.cpp

```diff
@@ -0,0 +1,200 @@
+          .planNode();
+
+  assertQuery(
+      plan,
+      "SELECT k1, k2, count(1), sum(a), max(b) FROM tmp GROUP BY CUBE (k1, k2)");
+}
+
+TEST_F(ExpandTest, rollup) {
+  vector_size_t size = 1'000;
+  auto data = makeRowVector(
+      {"k1", "k2", "a", "b"},
+      {
+          makeFlatVector<int64_t>(size, [](auto row) { return row % 11; }),
+          makeFlatVector<int64_t>(size, [](auto row) { return row % 17; }),
+          makeFlatVector<int64_t>(size, [](auto row) { return row; }),
+          makeFlatVector<std::string>(
+              size, [](auto row) { return std::string(row % 12, 'x'); }),
+      });
+
+  createDuckDbTable({data});
+
+  // Rollup.
+  auto plan =
+      PlanBuilder()
+          .values({data})
+          .expand(
+              {{"k1", "k2", "a", "b", "0 as group_id_0"},
+               {"k1", "null", "a", "b", "1"},
+               {"null", "null", "a", "b", "2"}})
+          .singleAggregation(
+              {"k1", "k2", "group_id_0"},
+              {"count(1) as count_1", "sum(a) as sum_a", "max(b) as max_b"})
+          .project({"k1", "k2", "count_1", "sum_a", "max_b"})
+          .planNode();
+
+  assertQuery(
+      plan,
+      "SELECT k1, k2, count(1), sum(a), max(b) FROM tmp GROUP BY ROLLUP (k1, k2)");
+}
+
+TEST_F(ExpandTest, countDistinct) {
+  vector_size_t size = 1'000;
+  auto data = makeRowVector(
+      {"a", "b"},
+      {
+          makeFlatVector<int64_t>(size, [](auto row) { return row; }),
+          makeFlatVector<std::string>(
+              size, [](auto row) { return std::string(row % 12, 'x'); }),
+      });
+  createDuckDbTable({data});
+
+  // count distinct.
+  auto plan =
+      PlanBuilder()
+          .values({data})
+          .expand({{"a", "null as b", "1 as group_id_0"}, {"null", "b", "2"}})
+          .singleAggregation(
+              //   {"group_id_0"},
+              {},
+              {"count(distinct a) as count_a", "count(distinct b) as count_b"})
+          .project({"count_a", "count_b"})
+          .planNode();
+
+  assertQuery(plan, "SELECT count(distinct a), count(distinct b) FROM tmp");
+}
+
+TEST_F(ExpandTest, invalidUseCases) {
+  vector_size_t size = 1'000;
+  auto data = makeRowVector(
+      {"k1", "k2", "a", "b"},
+      {
+          makeFlatVector<int64_t>(size, [](auto row) { return row % 11; }),
+          makeFlatVector<int64_t>(size, [](auto row) { return row % 17; }),
+          makeFlatVector<int64_t>(size, [](auto row) { return row; }),
+          makeFlatVector<std::string>(
+              size, [](auto row) { return std::string(row % 12, 'x'); }),
+      });
+
+  createDuckDbTable({data});
+
+  // Duplicate name check.
+  VELOX_ASSERT_USER_THROW(
+      PlanBuilder()
+          .values({data})
+          .expand(
+              {{"k1", "k1", "a", "b", "0 as group_id_0"},
+               {"k1", "null", "a", "b", "1"},
+               {"null", "null", "a", "b", "2"}})
+          .singleAggregation(
```

Comment:
Plan nodes past expand are not needed as we expect expand() call to throw. Please, update.